server/cmt_cmds.go

```diff
@@ -381,11 +386,19 @@ func BootstrapStateCmd[T types.Application](appCreator types.AppCreator[T]) *cob
 				height = app.CommitMultiStore().LastCommitID().Version
 			}
 
-			return node.BootstrapState(cmd.Context(), cfg, cmtcfg.DefaultDBProvider, uint64(height), nil)
+			var genProvider node.GenesisDocProvider
+			if legacyGenesis {
+				genProvider = node.DefaultGenesisDocProviderFunc(cfg)
+			} else {
+				genProvider = getGenDocProvider(cfg)
```

Comment:
We can directly just use this, no need for that `legacy-genesis` flag. Our function does automatically the fallback.

Reply:
awesome, cleaned up